src/panels/lovelace/common/validate-condition.ts

```diff
@@ -1,54 +1,156 @@
+import { HassEntity } from "home-assistant-js-websocket";
 import { ensureArray } from "../../../common/array/ensure-array";
 import { UNAVAILABLE } from "../../../data/entity";
 import { HomeAssistant } from "../../../types";
+import { isValidEntityId } from "../../../common/entity/valid_entity_id";
 
 export type Condition =
   | NumericStateCondition
-  | ScreenCondition
   | StateCondition
+  | ScreenCondition
   | UserCondition
   | OrCondition
   | AndCondition;
 
-export type LegacyCondition = {
+// Legacy conditional card condition
+export interface LegacyCondition {
   entity?: string;
   state?: string | string[];
   state_not?: string | string[];
-};
+}
+
+type FilterOperator =
+  | "=="
+  | "<="
+  | "<"
+  | ">="
+  | ">"
+  | "!="
+  | "in"
+  | "not in"
+  | "regex";
 
-export type NumericStateCondition = {
+// Legacy entity-filter badge & card condition
+export interface LegacyFilterCondition {
+  operator: FilterOperator;
+  entity?: string | number;
+  attribute?: string;
+  value: string | number | string[];
+}
+
+interface BaseCondition {
+  condition: string;
+}
+
+export interface NumericStateCondition extends BaseCondition {
   condition: "numeric_state";
   entity?: string;
-  below?: number;
-  above?: number;
-};
+  below?: string | number;
+  above?: string | number;
+}
 
-export type StateCondition = {
+export interface StateCondition extends BaseCondition {
   condition: "state";
   entity?: string;
-  state?: string | string[];
-  state_not?: string | string[];
-};
+  state?: string | number | string[];
+  state_not?: string | number | string[];
```

Comment:
State is always string, right?

Reply:
As entity-filter is not UI configurable, the user can put number in state.

But I can convert it in `buildConditionForFilter` if you want.

Reply:
fixed in e20bf741a4890f1b5c82794ede8ad1a235372345